frontend/src/scenes/experiments/components/SelectableCard.tsx

```diff
@@ -0,0 +1,62 @@
+import { IconCheckCircle } from '@posthog/icons'
+import { Tooltip } from 'lib/lemon-ui/Tooltip'
+import { cn } from 'lib/utils/css-classes'
+import { ReactNode } from 'react'
+
+interface SelectableCardProps {
+    title: string
+    description: ReactNode
+    selected: boolean
+    onClick: () => void
+    disabled?: boolean
+    disabledReason?: string
+    className?: string
+    'data-attr'?: string
+}
```

Comment:
With TS magic, we can do some enforcing here. For example, we can force people to add a reason when disabling the card by:

```ts
interface SelectableCardProps {
    title: string
    description: ReactNode
    selected: boolean
    onClick: () => void
    className?: string
    'data-attr'?: string
} & (
    | { disabled: true; disabledReason: string }
    | { disabled?: false; disabledReason?: never }
)
```

so TS will catch these edge cases:

```ts
// disabled but no reason
<SelectableCard disabled={true} />

// not disabled but reason provided
<SelectableCard disabled={false} disabledReason="Some reason" />
```